### ISSUE TITLE
Examples: ScoreCAM should not use the linear-activation model_modifier

### ISSUE THREAD

**Issue by bersbersbers** [keisen/tf-keras-vis#47]
https://arxiv.org/pdf/1910.01279.pdf says quite explicitly:

> The relative output value (postsoftmax) after normalization is more reasonable to measure the relevance than absolute output value (pre-softmax). Thus, in Score-CAM, we represent weight as post-softmax value, so that the score can be rescaled into a fixed range. [...] Score-CAM with softmax can well distinguish two different categories, even though the prediction probability of ‘cat’ is lower than the probability of ‘dog’. Normalization operation equips Score-CAM with good class discrimination ability

So I'd say in https://github.com/keisen/tf-keras-vis/blob/master/examples/attentions.ipynb, don't suggest use of `model_modifier` in [11] and [12]. 

See also algorithm 1, third line from the bottom, which may or may not imply *another* softmax step.

**Comment by keisen** [comment]
Hi, @bersbersbers .
Thank you for pointing that out! , and sorry for late reply.
As you said, the example has the problem, so I'm going to fix them by the end of this month.

Thanks!

**Comment by keisen** [comment]
Hi, @bersbersbers .
I'm facing a trouble unexpected.

I've modified the example to be NOT using the linear-activation model_modifier.
And then, although I've run the notebook, I got the image below I didn't expect.

![](https://github.com/keisen/tf-keras-vis/raw/release/v0.6.0/examples/images/score-cam-NOT-applied-model-modifier.png)

The heatmaps of Bear and Assault rifle is too noisy and not accurate.
When I again run GradCAM with the linear-activation model_modifier, I got it below.

![](https://github.com/keisen/tf-keras-vis/raw/release/v0.6.0/examples/images/score-cam-applied-model-modifier.png)

I'm NOT sure why it is exactly.
It may be any bugs of tf-keras-vis source code, the examination can take a time.
For now, I believe that , even if it is NOT the way correctly, ScoreCAM example should use the linear-activation model_modifier.


Thanks!

**Comment by bersbersbers** [comment]
I have also observed that ScoreCAMs get worse without the linear activation model modifier. However, I had a reason to keep this solution (without the modifier) since I am also interested in ScoreCAMs of classes other than the predicted one (let's say you want to know what you might need to change in the goldfish image to have it classified as bear - compare Fig. 5 of the ScoreCAM paper. Maybe you want to look at these ScoreCAMs as well.

In my case (VGG16, binary classification), I usually had the ScoreCAM of the wrong class being completely empty due to all scores being negative and the ReLU zeroing the resulting map. With one image, I even faced that situation with the correct class (predicted correctly with a softmax of 0.6/0.4) - the ScoreCAM of the *correct* class would be completely empty.

I agree that this may be worth additional investigation, as there may be some other things at play. Remember also that there may be another softmax step applied within the algorithm (not at the end of the network):

> See also algorithm 1, third line from the bottom, which may or may not imply _another_ softmax step.
